### ISSUE TITLE
input mask on readonly or disabled control prevents postback

### ISSUE THREAD

**Issue by tdinpsp** [RobinHerbots/Inputmask#2382]
I updated some libraries from GitHub (I didn't pay too much attention) however now any input control for which I have an input mask will not allow a postback (though it thinks it is doing a postback) if the control is either readonly or disabled.  It doesn't matter where on the page the control is or even if the control ever got focus from the user.

I'll be happy to try to provide more information if you have specific questions.

Thanks.

**Comment by tdinpsp** [comment]
I created a minimal Visual Studio 2019 project to demonstrate the issue.  I'm using .Net Framework 4.8 and the project file is attached.  When you run the MainPage you can click on the <Submit> button and it will show and update the time each time you click it.  If you then click the "Make Readonly" button then the "Submit" button will no longer work.  If you then click "Make Editable" then the "Submit" button will work again.

[InputMaskProblem.zip](https://github.com/RobinHerbots/Inputmask/files/5134597/InputMaskProblem.zip)

**Comment by bodelia** [comment]
Confirming issue.

Might be intended to prevent form from submitting if  fields are disabled or readonly but then js should  throw an Exception.
Spent half a day to find where the problem came from.

Should either be more explicit on the submit prevention or permit submission.

here is a temporary solution :

```
                $form.on('submit', function () {
                    $.each($('.mask-currency-euro'), function (key, input) {
                        $(input).attr('disabled', false);
                        // Either one
                        $(input).attr('readonly', false);
                    });
                })
```
hope it helps :)

**Comment by RobinHerbots** [comment]
@tdinpsp , @walsyphtis , 

The issue is fixed in version 5.0.6-beta.15

Can you have a try.

**Comment by bodelia** [comment]
@RobinHerbots 
Seems to work perfectly. 
Thank you for your work !
